src/System.Net.Primitives/tests/FunctionalTests/IPEndPointTest.cs

```diff
@@ -2,126 +2,262 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Collections.Generic;
+using System.Net.Sockets;
 using Xunit;
 
 namespace System.Net.Primitives.Functional.Tests
 {
-    public static class IPEndPointTest
+    public class IPEndPointTest
```

Comment:
You don't seem to be deriving from this type, or adding setup state, or adding logging scopes; why un-static the test class?

Reply:
I recall that most test classes aren't static and just wanted to increase consistency while i was here. IMO the addition of `static` just makes the test name look longer and makes it harder for me to see long test names

Reply:
> I recall that most test classes aren't static

I can't speak to the codebase as a whole, but the ones I've written (and most of the ones I feel I've reviewed) are static unless they need to not be for one of the reasons I listed.

So for consistency and minimizing noise in the diff, please restore this class to static.

Reply:
> I can't speak to the codebase as a whole, but the ones I've written (and most of the ones I feel I've reviewed) are static unless they need to not be for one of the reasons I listed.

Not entirely scientific, but:
```C#
using System;
using System.IO;

class Program
{
    static void Main()
    {
        int instanceMethod = 0;
        int staticMethod = 0;

        foreach (string path in Directory.EnumerateFiles(@"d:\repos\corefx\src\", "*.cs", SearchOption.AllDirectories))
        {
            if (!path.Contains(@"\tests\")) continue;

            string[] lines = File.ReadAllLines(path);
            foreach (string line in lines)
            {
                if (line.Contains("public void ") || line.Contains("public Task ")) instanceMethod++;
                else if (line.Contains("public static void ") || line.Contains("public static Task ")) staticMethod++;
            }
        }

        Console.WriteLine("Instance: " + instanceMethod);
        Console.WriteLine("Static  : " + staticMethod);
    }
}
```
outputs:
```
Instance: 38824
Static  : 24441
```

> minimizing noise in the diff

I agree with that.  We shouldn't change it unless we need to; otherwise it's just going to shift back and forth based on the current dev's personal preference.

Separetely it'd be nice to come up with a guideline, change the codebase to match it, and then stick with it.

Reply:
>Separately it'd be nice to come up with a guideline, change the codebase to match it, and then stick with it. My preference is: there's little benefit to making them static, and there are reasons tests may need to be instance, so best to just have everything be instance.

I agree that we should have a guideline. And I also agree that the tests should be instance and not static (unless required).

In the interests of moving this PR along, it would be best to focus on the specific issues raised in this PR regarding the product code:

* A `Debug.Assert` that seems invalid and should be removed
* Adding the throwing of an Argument*Exception for certain inputs. And that potentially is a breaking change (which might be fine for a major release like 5.0).

@hughbe Can you please revise this PR to focus on just these points? That would make the PR easier to review and finish.  Thanks!